### PR TITLE
[EDA] Affiliations Created with Program Enrollments Page Section and Role for Created Affiliations

### PR DESCRIPTION
**Pull request by skosgi1**
# Critical Changes

# Changes

# Issues Closed

# New Metadata
**Apex Classes:**
AffiliationsWithProgramEnrollmentModel.cls
AffiliationsWithProgramEnrollmentMapper.cls
AffiliationsWithProgramEnrollmentService.cls
AffiliationsWithProgramEnrollmentVModel.cls
AffiliationsWithProgramEnrollmentVMapper.cls
AffiliationsWithProgramEnrollmentService.cls
AffiliationsWithProgramEnrollController.cls
AffiliationsWithPrgrmEnrolCtrl_TEST.cls
AffiliationsWithPrgrmEnrolVMapr_TEST.cls
AffiliationsWithPrgrmEnrolVModel_TEST.cls
AffiliationsWithPrgrmEnrolSrvc_TEST.cls
AffiliationsWithPrgrmEnrolMapr_TEST.cls
AffiliationsWithPrgrmEnrolModel_TEST.cls

**Custom Labels:**
stgAfflProgramEnrollmentSettingsTitle

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: force-app/main/default/lwc/programSettings/programSettings.js [hunk 1]
```diff
@@ -54,6 +97,90 @@ export default class programSettings extends LightningElement {
...
+
+        if (result.data) {
+            this.affiliationsWithProgramEnrollmentVModel = result.data;
+        } else if (result.error) {
+            //console.log("error retrieving preferredContactInfoSettingsVModel");
+        }
+    }
+
+    handleRoleForCreatedAfflChange(event) {
+        this.handleSpecifyRoleForCreatedAffiliaitons(event);
```
Comment (skosgi1) [inline force-app/main/default/lwc/programSettings/programSettings.js:112]: @MatthewBlanski is this the right way to handle the Specify Role scenario?

File: force-app/main/default/classes/AffiliationsWithPrgrmEnrolVMapr_TEST.cls [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    /***********************************************************************************************************************************
+     ****************************************************** STUBS ***********************************************************************
+     ***********************************************************************************************************************************/
+
+    /**************************************************************************************************************************************
+     * @description Stub class to simulate the response from PicklistEntryService to return a valid respose
+     **************************************************************************************************************************************/
+    private class STUB_PicklistEntryService extends PicklistEntryService {
```
Comment (skosgi1) [inline force-app/main/default/classes/AffiliationsWithPrgrmEnrolVMapr_TEST.cls:186]: @MatthewBlanski I am using only 1 stub for 2 methods calls, is this ok?

Comment (MatthewBlanski) [reply force-app/main/default/classes/AffiliationsWithPrgrmEnrolVMapr_TEST.cls:186]: This should be fine. Stub where stubs are needed for unit testing.

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]
Please add the new items to the nav. Everything else checks out.

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]
My mistake on this, code coverage ended out at 100% on a second run. It looks like some tests failed to run on the first one. We still need the nav updates.

File: force-app/main/default/lwc/edaSettingsNavigation/edaSettingsNavigation.js [hunk 3]
```diff
@@ -82,8 +88,12 @@ export default class EdaSettingsNavigation extends LightningElement {
         },
         programSettings: {
             programSettings: stgProgramsSettingsNav,
-            programAutoEnrollmentMappings: "Program Auto-Enrollment Mappings",
+            programAutoEnrollmentMappings: stgAutoEnrollmentProgramTitle,
             programEnrollmentDeletions: "Program Enrollment Deletions",
```
Comment (MatthewBlanski) [inline force-app/main/default/lwc/edaSettingsNavigation/edaSettingsNavigation.js:92]: This is still hardcoded. Can you tend to this as well? We should have a label for this.

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]

**Review by MatthewBlanski — APPROVED** [review]

File: force-app/main/default/classes/AffiliationsWithProgramEnrollmentVMapper.cls [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        Map<String, PicklistEntryModel> allAffiliationRolePicklistEntryModelsByName = this.locatePicklistEntryService()
+            .getModelsByNameForSObjectField(Affiliation__c.SObjectType, Affiliation__c.Role__c);
+
+        Map<String, PicklistEntryModel> activeAffiliationRolePicklistEntryModelsByName = this.locatePicklistEntryService()
+            .getActivePicklistEntryModelsByName(allAffiliationRolePicklistEntryModelsByName);
+
+        List<ComboboxValueVModel> affiliationRoleComboboxValueVModelsList = new List<ComboboxValueVModel>();
+
+        // add None option, maps to a blank value in hierarchy settings field
+        affiliationRoleComboboxValueVModelsList.add(new ComboboxValueVModel(label.NoneOption, '""'));
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AffiliationsWithProgramEnrollmentVMapper.cls:80]: Hold on to this value s we can override to it later.

File: force-app/main/default/classes/AffiliationsWithProgramEnrollmentVMapper.cls [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+        for (PicklistEntryModel picklistEntryModel : activeAffiliationRolePicklistEntryModelsByName.values()) {
+            affiliationRoleComboboxValueVModelsList.add(
+                new ComboboxValueVModel(picklistEntryModel.label, picklistEntryModel.name)
+            );
+        }
+
+        ComboboxVModel affiliationRoleComboboxVModel = new ComboboxVModel(
+            currentAffiliationsWithProgramEnrollmentModel.role,
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AffiliationsWithProgramEnrollmentVMapper.cls:89]: We need to have an override for this to go to "None" if Specify Role is false.

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 6]
```diff
@@ -886,14 +886,6 @@
         <shortDescription>Label for the Affiliations Related to Deleted Program Enrollments setting</shortDescription>
         <value>Affiliations Related to Deleted Program Enrollments</value>
     </labels>
-    <labels>
-        <fullName>stgAfflProgEnrollDeleteTitle</fullName>
```
Comment (MatthewBlanski) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:890]: Why did we delete this label? We are still referencing it.

**Review by MatthewBlanski — APPROVED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 6 above)
Comment (skosgi1) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:890]: resolved offline.

**Comment by spinkelman** [comment]
QA passes - Merging

**Pull request merged**